sky/backends/backend_utils.py

```diff
@@ -371,6 +371,10 @@ def _get_generated_config(cls, autogen_comment: str, host_name: str,
             proxy = f'ProxyCommand {proxy_command}'
         else:
             proxy = ''
+        # StrictHostKeyChecking=no skips the host key check for the first
+        # time. UserKnownHostsFile=/dev/null prevents the host key from being
+        # added to the known_hosts file, so that the host key check is skipped
+        # for all subsequent connections.
```

Comment:
In addition to not allowing adding of the host_key, I think setting `UserKnownHostsFile=/dev/null` also always returns an empty file for known hosts, thus making SSH think this is a first-time connection and thus skipping the host key check. Might be worth mentioning in the comment

Reply:
Good point! Added. Thanks!